src/utils/HashUtils.ts

```diff
@@ -177,25 +177,11 @@ export class HashUtils {
 
   public static encodeObject(obj: Object): string {
     const retArray = _.map(<_.Dictionary<any>>obj, (val, key?, obj?) => {
-      return `"${Utils.safeEncodeURIComponent(key)}":${this.encodeValue(val)}`;
+      return `"${Utils.safeEncodeURIComponent(key)}":${JSON.stringify(val)}`;
```

Comment:
It's kind of a weird change.

What wasn't working with the other method ?

Reply:
Was not working with nested objects, like the `fv` state:

`fv={"key": ["value1","value2"]}` would be serialized as `fv={"key":[value1, value2]}` which would fail at deserialization (missing quotes).

See related unit test :)